teachopencadd/talktorials/T014_binding_site_detection/talktorial.ipynb

```diff
@@ -0,0 +1,2116 @@
+{
```

Comment:
<p>Have you checked if this code is as simple as it can be? If not, I can look into this a bit more.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/volkamerlab/teachopencadd/pull/86/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

Reply:
<p>@dominiquesydow </p><p>The current version was inherited from the internship's and Jaime's version. I felt it was easy enough to follow given the detailed comments on what is done in each step, but happy if you have some ideas on how to simplify it.</p>

Reply:
Wow, all this

```python
# Split string into list of lists (= table)
result_table_split = [i.split('\t') for i in result_table[:-1].split('\n')]

# Remove spaces
result_table_split = [[j.replace(' ', '') for j in i] for i in result_table_split]

# Extract column names, index names, table body
column_names = result_table_split[0]
index_names = [i[0] for i in result_table_split[1:]]
table = [i[1:] for i in result_table_split[1:]]

# Convert to DataFrame
result_table_df = pd.DataFrame(
    table,
    columns=column_names[1:],
    index=index_names
)
result_table_df.index.name = 'name'

# Convert numbers from strings to numeric values
for name, data in result_table_df.iteritems():
    try:
        result_table_df[name] = pd.to_numeric(data)
    except ValueError:
        pass
```
can be reduced to 

```python
import io
result_table_df = pd.read_csv(io.StringIO(result_table), sep="\t").set_index("name")
```
because `pandas` handles all the line and separator splitting + casts columns automatically to their correct dtypes.

I love this :)

Reply:
Wow this is amazing!!!

Reply:
When you include it can you add a short comment about the *magic* it does?

Reply:
Good point, added
```python
    # Load the table (csv format using "\t" as separator) with pandas DataFrame
    # We cannot load the table from a string directly but from a file
    # Use io.StringIO to pass this string in file format as needed for read_csv method
    result_table_df = pd.read_csv(io.StringIO(result_table), sep="\t").set_index("name")
```

Reply:
Nitpicking here but "file format" is not entirely accurate. I'd say something like this:

```
# Use io.StringIO to wrap this string a file-like object as needed for read_csv method
# See more: https://docs.python.org/3/library/io.html#io.StringIO
```

Reply:
Not nitpicky at all :) Thanks, will update this!